### PR TITLE
Feature/add mongodb

### PR DESCRIPTION
**Pull request by basti-n**
some very basic setup for mongodb, with three collections:

1. Base Archie Data (date of birth, weight, etc)
2. Food Data (data about the energy and macro nutrient characteristics of food)
3. Food Intake Data (tracking data of what Archie has eaten)

### CODE REVIEW

File: apps/api/src/app/app.module.ts [hunk 1]
```diff
@@ -1,10 +1,41 @@
 import { Module } from '@nestjs/common';
-
 import { AppController } from './app.controller';
 import { AppService } from './app.service';
+import { MongooseModule } from '@nestjs/mongoose';
+import { ConfigService, ConfigModule } from '@nestjs/config';
+import * as Joi from 'joi';
+import { FoodModule } from './food/food.module';
+import { IntakeModule } from './intake/intake.module';
+import { ArchieModule } from './archie/archie.module';
```
Comment (miriamhaenle) [inline apps/api/src/app/app.module.ts:9]:
```suggestion
import { MongooseModule } from '@nestjs/mongoose';
import { ConfigService, ConfigModule } from '@nestjs/config';
import * as Joi from 'joi';

import { AppController } from './app.controller';
import { AppService } from './app.service';
import { ArchieModule } from './archie/archie.module';
import { FoodModule } from './food/food.module';
import { IntakeModule } from './intake/intake.module';
```

File: apps/api/src/app/app.service.ts [hunk 2]
```diff
@@ -5,6 +5,11 @@ import { IArchie } from '@archmaster/shared/types';
 export class AppService {
   getArchie(): IArchie {
```
Comment (miriamhaenle) [inline apps/api/src/app/app.service.ts:6]: Maybe we should use a more generic name like `getCuteDog` 😛  `getDog`

File: apps/api/src/app/db/schema/archie.schema.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { IFoodHistory, IArchie } from '@archmaster/shared/types';
+import { FoodEntrySchema } from './food-entry.schema';
+
+export type ArchieDocument = Document & Archie;
```
Comment (miriamhaenle) [inline apps/api/src/app/db/schema/archie.schema.ts:6]: `Document` is missing

File: apps/api/src/app/app.service.ts (hunk 2 above)
Comment (basti-n) [reply apps/api/src/app/app.service.ts:6]: we can change it when implementing the controllers and services. Which in turn will depend on how the contract between FE and BE should look like.

File: apps/api/src/app/db/schema/archie.schema.ts (hunk 3 above)
Comment (basti-n) [reply apps/api/src/app/db/schema/archie.schema.ts:6]: True, updated.

File: apps/api/src/app/app.module.ts (hunk 1 above)
Comment (basti-n) [reply apps/api/src/app/app.module.ts:9]: so dependency imports before local/file imports? Can we add a linting rule for it?

Comment (miriamhaenle) [reply apps/api/src/app/app.module.ts:9]: Yes, lets add a linting rule in a separate PR

**Pull request merged**